internal/bloblang/query/functions.go

```diff
@@ -699,6 +700,24 @@ var _ = registerSimpleFunction(
 
 //------------------------------------------------------------------------------
 
+var _ = registerSimpleFunction(
+	NewFunctionSpec(
+		FunctionCategoryGeneral, "snowflake_id",
+		"Generate a new snowflake ID each time it is invoked and prints a string representation.",
+		NewExampleSpec("", `root.id = snowflake_id()`),
+	),
+	func(_ FunctionContext) (interface{}, error) {
+		node, err := snowflake.NewNode(1)
+		if err != nil {
+			panic(err)
```

Comment:
We'd rather avoid panicking here if possible. Please have a look at other functions for error handling patterns.

Reply:
Thanks for the suggestion, it's done ✅